src/index.js

```diff
@@ -67,29 +76,86 @@ export function getTagValue(list, name) {
 		return;
 	}
 
-	let jwk = JSON.parse(Buffer.from(DEPLOY_KEY, 'base64').toString('utf-8'));
+	// Throw an error if both --eth and --pol are true
+	if (argv.eth && argv.pol) {
+		console.error('Error: Cannot deploy with both ETH and POL.');
+		process.exit(1); // Exit with an error code
+	}
+
+	let jwk;
+	if (argv.pol || argv.eth) jwk = DEPLOY_KEY;
+	else {
+		jwk = JSON.parse(Buffer.from(DEPLOY_KEY, 'base64').toString('utf-8'));
+	}
+
 	try {
-		const manifestId = await TurboDeploy(argv, jwk);
-		const signer = new ArweaveSigner(jwk);
+		let signer;
+		let token;
+
+		// Creates proper signer based on wallet type.
+		switch (true) {
+			case argv.eth:
+				signer = new EthereumSigner(jwk);
+				token = 'ethereum';
+				break;
+			case argv.pol:
+				signer = new EthereumSigner(jwk);
+				token = 'pol';
+				break;
+			default:
+				signer = new ArweaveSigner(jwk);
+				token = 'arweave';
+				break;
+		}
+
+		const turbo = TurboFactory.authenticated({
+			signer: signer,
+			token: token,
+		});
+
+		const uploadResult = await turbo.uploadFolder({
+			folderPath: argv['deploy-folder'],
+			dataItemOpts: {
+				tags: [
+					{
+						name: 'App-Name',
+						value: 'Permaweb-Deploy',
+					},
+					// prevents identical transaction Ids from eth wallets
+					{
+						name: 'anchor',
+						value: new Date().toISOString(),
+					},
+				],
+			},
+		});
+
+		const manifestId = uploadResult.manifestResponse.id;
+
 		const ant = ANT.init({ processId: ANT_PROCESS, signer });
 
-		// Update the ANT record (assumes the JWK is a controller or owner)
+		// Update the ANT record (assumes the signer is a controller or owner)
 		await ant.setRecord(
 			{
 				undername: argv.undername,
 				transactionId: manifestId,
 				ttlSeconds: 3600,
-			},{
+			},
+			{
 				tags: [
 					{
 						name: 'GIT-HASH',
-						value: process.env.GITHUB_SHA,
+						value: process.env.GITHUB_SHA || '',
 					},
 					{
 						name: 'App-Name',
 						value: 'Permaweb-Deploy',
 					},
-				]
+					{
+						name: 'anchor',
+						value: new Date().toISOString(),
+					},
+				],
```

Comment:
Could add options for custom tags in the cli here

Reply:
Agreed, but this is outside the scope of this pr.

Reply:
no need to add anchor here - the github sha will be unique - and i believe we add this anchor in the ario sdk already